packages/material-ui-codemod/README.md

```diff
@@ -44,6 +44,36 @@ This codemod tries to perform a basic expression simplification which can be imp
 // Limitation
 -const spacing = theme.spacing.unit * 5 * 5;
 +const spacing = theme.spacing(5) * 5;
+
+#### `optimal-imports`
+
+Converts all `@material-ui/core` imports more than 1 level deep to the optimal
+form for tree shaking:
+
+```diff
+-import withStyles from '@material-ui/core/styles/withStyles';
+-import createMuiTheme from '@material-ui/core/styles/createMuiTheme';
++import {withStyles, createMuiTheme} from '@material-ui/core/styles';
+```
+
+```sh
+find src -name '*.js' -print | xargs jscodeshift -t node_modules/@material-ui/codemod/lib/v4.0.0/optimal-imports.js
+```
+
+### v3.0.0
+
+#### `top-level-imports`
```

Comment:
I'd like both of them to have the same name i.e. `optimal-imports`. The desired outcome is the same but the implementation
can be different in different majors.

Reply:
Now that I think about it, is the v3 one actually optimal for v3?  If there's no tree shaking in v3, doesn't that mean importing from top level will cause all of material-ui's code to get bundled?

As far as I understood the docs, with v4 one can either import from top level or second level and tree shaking should work, so maybe they are both different options for v4?

Reply:
There's tree shaking for top level. Just not second level.

Reply:
Okay, @oliviertassinari the blanket statement that "tree shaking doesn't work in v3" confused me

Reply:
I have never seen it work, even for the top level in v3. But I would rather trust @eps1lon than what I have experienced. I have never dived into it.